synapse/handlers/sync.py

```diff
@@ -94,7 +94,12 @@ def __nonzero__(self) -> bool:
     __bool__ = __nonzero__  # python3
 
 
-@attr.s(slots=True, frozen=True)
+# We can't freeze this class, because we need to update it after it's instantiated to
+# update its unread count. This is because we calculate the unread count for a room only
+# if there are updates for it, which we check after the instance has been created.
+# This should not be a big deal because we update the notification counts afterwards as
+# well anyway.
+@attr.s(slots=True, frozen=False)
```

Comment:
this is the default
```suggestion
@attr.s(slots=True)
```

Reply:
Right, I wanted to make it explicit but it's probably unnecessary.